vaadin-testbench-shared/src/main/java/com/vaadin/testbench/ElementQuery.java

```diff
@@ -138,69 +144,242 @@ public ElementQuery(Class<T> elementClass, String tagName) {
     }
 
     /**
-     * Executes a search for element with the given id.
+     * Selects on elements having the given attribute.
+     * <p>
+     *     Note, this attribute need not have a value.
+     * </p>
      *
-     * @param id
-     *            the id to look up
-     * @return the element with the given id
+     * @param name
+     *            the attribute name
+     * @return this element query instance for chaining
      *
-     * @throws NoSuchElementException
-     *             if no element is found
+     * @deprecated use {@link #withAttribute}
      */
-    public T id(String id) {
-        return attribute("id", id).first();
+    @Deprecated(since = "9.2")
```

Comment:
I suppose this change will go to the next minor
```suggestion
    @Deprecated(since = "9.3")
```